api/v1alpha2/linodecluster_types.go

```diff
@@ -128,6 +128,9 @@ type NetworkSpec struct {
 	// If not set, defaults to 30
 	// +optional
 	DNSTTLSec int `json:"dnsTTLsec,omitempty"`
+	// DNSEndpointOverride is used to override CAPL's construction of the controlplane endpoint
+	// +optional
+	DNSEndpointOverride string `json:"dnsEndpointOverride,omitempty"`
```

Comment:
Could we maybe rename this to just `DNSEndpoint` with a description of it overriding the other settings? also might be worth in this message saying specifically it ignores roodDomain+unique identifier

Reply:
renamed and updated the comment